### PR TITLE
resilient piping failover

### PR DESCRIPTION
**Pull request by rynomad**

### CODE REVIEW

File: packages/gateway-client/src/worker/index.ts [hunk 1]
```diff
@@ -893,17 +933,28 @@ async function main() {
     node.connectionManager.addEventListener('peer:disconnect', evt => {
         const connection = evt.detail;
         console.log(`Disconnected from ${connection.remotePeer.toString()}`);
+        if (connection.remotePeer.equals(self.serverPeer)) {
+            console.log('disconnected from server');
+            // update status
+            self.status.serverPeer = ServerPeerStatus.CONNECTING;
```
Comment (JoshuaCWebDeveloper) [inline packages/gateway-client/src/worker/index.ts:939]: Nice touch.

File: packages/gateway-client/src/worker/index.ts [hunk 2]
```diff
@@ -204,6 +204,7 @@ self.localforage = localforage;
 
 async function getWSOpenLatency(ma: string): Promise<number> {
     return new Promise(resolve => {
+        setTimeout(resolve, 5000, Infinity);
```
Comment (JoshuaCWebDeveloper) [inline packages/gateway-client/src/worker/index.ts:207]: I would recommend going back to 1 second.

File: packages/gateway-client/src/worker/index.ts [hunk 3]
```diff
@@ -769,28 +796,41 @@ async function getBootstrapList(skipFetch = false) {
...
     self.status.relays.push(...bootstraplist.slice(2));
 
-    const datastore = new LevelDatastore('./libp2p');
-    await datastore.open(); // level database must be ready before node boot
+    // const datastore = new LevelDatastore('./libp2p');
+    // await datastore.open(); // level database must be ready before node boot
     const serverID = bootstraplist[0].split('/').pop();
     const node = await createLibp2p({
-        datastore,
+        // datastore,
```
Comment (JoshuaCWebDeveloper) [inline packages/gateway-client/src/worker/index.ts:833]: Are we using some sort of default datastore now?

File: packages/gateway-client/src/worker/index.ts [hunk 4]
```diff
@@ -221,22 +222,23 @@ async function getWSOpenLatency(ma: string): Promise<number> {
...
-async function initCheckAddresses(addresses: string[]): Promise<void> {
+async function initCheckAddresses(addresses: string[]): Promise<string[]> {
     self.latencyMap = new Map();
     self.latencySet = new Set();
-    await Promise.race([
-        Promise.all(addresses.map(checkAddress)),
-        waitFor(1000),
-    ]);
+    await Promise.all(addresses.map(checkAddress));
+    return addresses.filter(a => self.latencyMap.has(a));
```
Comment (JoshuaCWebDeveloper) [inline packages/gateway-client/src/worker/index.ts:241]: Smart.

File: packages/gateway-client/src/worker/index.ts [hunk 5]
```diff
@@ -844,12 +884,12 @@ async function main() {
                 async evt => {
                     const connection = evt.detail;
                     const str_id = connection.remotePeer.toString();
-                    console.log(`Connected to ${str_id}, check ${serverID}`);
```
Comment (JoshuaCWebDeveloper) [inline packages/gateway-client/src/worker/index.ts:847]: FYI, I'm actually adding more logging here not less in my refactor, this info is pretty useful.

**Review by JoshuaCWebDeveloper — APPROVED** [review]

File: packages/gateway-client/src/worker/index.ts (hunk 3 above)
Comment (rynomad) [reply packages/gateway-client/src/worker/index.ts:833]: in-memory, will get reset every time the script restarts, but we're explicitely keeping track of our own relays.

**Pull request merged**